modules/images/image-loading-optimization/optimization.php

```diff
@@ -114,24 +96,29 @@ function ilo_optimize_template_output_buffer( string $buffer ): string {
 	$slug = ilo_get_url_metrics_slug( ilo_get_normalized_query_vars() );
 	$post = ilo_get_url_metrics_post( $slug );
 
-	// No URL metrics are present, so there's nothing we can do.
-	if ( ! $post ) {
-		return $buffer;
-	}
+	$url_metrics = $post ? ilo_parse_stored_url_metrics( $post ) : array();
```

Comment:
Why remove the early return here? Isn't that more efficient if no URL metrics post is found?

Reply:
Good question. Because now if no URL metrics are found, we need to proceed and add breadcrumbs to the elements so that we can gather the URL metrics. You can see below too that the detection script is now injected in this function as well.